server/container/util/src/test/java/org/apache/james/util/SizeFormatTest.java

```diff
@@ -86,4 +86,64 @@ void formatShouldUseTiBWhenExactlyOneTiB() {
             .isEqualTo("1 TiB");
     }
 
+    @Test
+    void parseAsByteCountShouldReturnCountWhenNoUnit() {
+        assertThat(SizeFormat.parseAsByteCount("36"))
+            .isEqualTo(36);
+    }
+
+    @Test
+    void parseAsByteCountShouldAcceptKiB() {
+        assertThat(SizeFormat.parseAsByteCount("36 KiB"))
+            .isEqualTo(36 * 1024);
+    }
+
+    @Test
+    void parseAsByteCountShouldAcceptZero() {
+        assertThat(SizeFormat.parseAsByteCount("0 KiB"))
+            .isEqualTo(0);
+    }
+
+    @Test
+    void parseAsByteCountShouldThrowOnInvalidUnit() {
+        assertThatThrownBy(() -> SizeFormat.parseAsByteCount("0 invalid"))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void parseAsByteCountShoulfThrowOnMissingAmount() {
+        assertThatThrownBy(() -> SizeFormat.parseAsByteCount("KiB"))
+            .isInstanceOf(NumberFormatException.class);
+    }
+
+    @Test
+    void parseAsByteCountShouldThrowWhenEmpty() {
+        assertThatThrownBy(() -> SizeFormat.parseAsByteCount(""))
+            .isInstanceOf(NumberFormatException.class);
+    }
+
+    @Test
+    void parseAsByteCountShouldThrowWhenUnitDoNotMatchCase() {
+        assertThatThrownBy(() -> SizeFormat.parseAsByteCount("12 KIB"))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void parseAsByteCountShouldAcceptNegativeValue() {
+        assertThat(SizeFormat.parseAsByteCount("-36 KiB"))
+            .isEqualTo(-36 * 1024);
```

Comment:
Should those `36 * 1024` are extract to variable?

Reply:
In my opinion a constant extraction here would harm readability: I prefer having a raw amount here.